sky/authentication.py

```diff
@@ -401,4 +404,47 @@ def setup_kubernetes_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
             logger.error(suffix)
             raise
 
+    sshjump_name = clouds.Kubernetes.SKY_SSH_JUMP_NAME
+    sshjump_image =  clouds.Kubernetes.SSH_JUMP_IMAGE
+
+    template_path = os.path.join(sky.__root_dir__, 'templates', 'kubernetes-sshjump.j2')
+    if not os.path.exists(template_path):
+        raise FileNotFoundError('Template "kubernetes-sshjump.j2" does not exist.')
+    with open(template_path) as fin:
+        template = fin.read()
+    j2_template = jinja2.Template(template)
+    _content = j2_template.render(name=sshjump_name, image=sshjump_image, secret=key_label)
+
+    content = json.loads(_content)
+
+    try:
+        kubernetes.core_api().create_namespaced_pod('default', content['pod_spec'])
```

Comment:
We need to think a bit about the lifecycle of this jump pod. 

* **Creation** - Creation of this jump ssh pod happens when the user creates their first SkyPilot cluster in the Kubernetes cluster. 
* **Run** - The jump SSH pod is kept alive as long as there is a SkyPilot cluster running (i.e., there exists a Kubernetes pod with label `parent: skypilot` in the same namespace)
* **Termination** - If there hasn't been a pod with label `parent: skypilot` in the same namespace since the past n minutes (say, 10 minutes), this jump pod should terminate itself.

We can do this by running a script like this in the jump ssh pod as the `command` for the pod:

```python
import os
import datetime
import pytz
import time
from kubernetes import client, config

# Load kube config
config.load_incluster_config()

v1 = client.CoreV1Api()

# Get the current namespace from the pod service account
with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r") as f:
    current_namespace = f.read()

# Set the time delta for checking last active pods
time_delta = datetime.timedelta(minutes=10)

# Set delay for each retry
retry_delay = 60  # In seconds

while True:
    # Get the current time
    now = datetime.datetime.now(pytz.UTC)

    found = False
    # List the pods in the current namespace
    ret = v1.list_namespaced_pod(current_namespace)
    for i in ret.items:
        if i.metadata.labels and 'parent' in i.metadata.labels and i.metadata.labels['parent'] == 'skypilot':
            # Calculate the elapsed time since the pod was last active
            elapsed_time = now - i.metadata.creation_timestamp
            # If the pod was active in the last 10 minutes, set found to True
            if elapsed_time < time_delta:
                found = True
                break

    # If no active pods were found with the specified label, exit the script
    if not found:
        print("No active pods found with label 'parent: skypilot' in the past 10 minutes. Exiting...")
        exit(1)

    # If pods were found, sleep for the specified delay and then retry
    print(f"Active pods found with label 'parent: skypilot'. Retrying in {retry_delay} seconds...")
    time.sleep(retry_delay)
```

What do you think?